WordPress/src/main/java/org/wordpress/android/ui/utils/JetpackAppMigrationFlowUtils.kt

```diff
@@ -30,11 +36,18 @@ class JetpackAppMigrationFlowUtils @Inject constructor(
             && !appPrefsWrapper.isJetpackMigrationCompleted()
             && isWordPressInstalled()
             && isWordPressCompatible()
+            && isMigrationEligible()
```

Comment:
Is it necessary to invoke the eligibility check here?

The main concern I have about this is that `shouldShowMigrationFlow` is usually called from the UI thread, and this has the potential be somewhat IO-heavy. This is usually part of the migration itself, since behind the scenes, it is using a content resolver to [query all post and page content on every site](https://github.com/wordpress-mobile/WordPress-Android/blob/a01543b9d72732e07fe2b43bf479ba90fb2a24cd/WordPress/src/main/java/org/wordpress/android/localcontentmigration/LocalEligibilityStatusProviderHelper.kt#L26) in the WordPress app.

We have the `isJetpackMigrationEligible()` flag in the prefs, but I imagine that this was added here because this is needed for the first go around (let me know if that is a wrong assumption :sweat_smile: ). To handle that scenario, I wonder if we could somehow utilize the `Ineligible` state transition [in the viewModel](https://github.com/wordpress-mobile/WordPress-Android/blob/536e24f4c2f87ad4c0b0e7007504815be96549d0/WordPress/src/main/java/org/wordpress/android/ui/main/jetpack/migration/JetpackMigrationViewModel.kt#L95)? Wdyt?

Reply:
> Is it necessary to invoke the eligibility check here?

Good question! 👍 

> We have the `isJetpackMigrationEligible()` flag in the prefs, but I imagine that this was added here because this is needed for the first go around (let me know if that is a wrong assumption 😅 ).

It was indeed the case, without it the flow via deep link was also not working correctly in situations where:
1. The WordPress app is not logged in → JP Migration error `WPNotLoggedIn`
  In this case the desired functionality of opening the Jetpack app from links should be the same as without this PR, the Login screen should be shown and the user would end up on the correct page after the login.
2. There are local draft changes in the WordPress app not synced with the backend → Error `LocalDraftContentIsPresent`.

> To handle that scenario, I wonder if we could somehow utilize the Ineligible state transition [in the viewModel](https://github.com/wordpress-mobile/WordPress-Android/blob/536e24f4c2f87ad4c0b0e7007504815be96549d0/WordPress/src/main/java/org/wordpress/android/ui/main/jetpack/migration/JetpackMigrationViewModel.kt#L95)? Wdyt?

I already explored this solution to some extent, but it wasn't producing the desired result (see point 1 above). I'm not sure anymore what wasn't ok, but I can look more into it 👍